pkg/queue/sharedmain/main.go

```diff
@@ -304,18 +307,19 @@ func Main(opts ...Option) error {
 		}
 		logger.Info("Received TERM signal, attempting to gracefully shutdown servers.")
 		logger.Infof("Sleeping %v to allow K8s propagation of non-ready state", drainSleepDuration)
-		drainer.Drain()
-
-		// Removing the main server from the shutdown logic as we've already shut it down.
-		delete(httpServers, "main")
-
-		for serverName, srv := range httpServers {
-			logger.Info("Shutting down server: ", serverName)
-			if err := srv.Shutdown(context.Background()); err != nil {
-				logger.Errorw("Failed to shutdown server", zap.String("server", serverName), zap.Error(err))
+		drain(drainer, drainerTLS)
```

Comment:
@dprotaso 
We had 2 drainers before and we have 2 drainers now. 
Previously we mistreated the second drainer and did not drain it which resulted in the code not working as it was designed. Now we solved this problem by draining both brainers.

Reply:
> Now we solved this problem by draining both drainers.

A queue proxy should only have a single drainer

Reply:
Since we now have separate http handlers for the separate main servers (HTTP and HTTPS), this results in two drainers. This was not introduced by this PR, it is the way the code works today.  

My understanding is that if/when we will have a PR that will merge the two HTTP handlers into one (and I am not presently sure what implications this may have), we can then go to a single drainer we well.  

Right now, my understanding is that as long as we keep the two servers running on a separate set of HTTP handlers, we also internationally create a drainer for each one and with this PR we also drain those two drainers i.e. we fix a bug from a previous PR that ignored the second drainer and did not drain it which afaik was not intentional and as far as my tests have shown resulted in wrong behavior.